src/explorer/DeploymentSlotsNode.ts

```diff
@@ -60,4 +61,50 @@ export class DeploymentSlotsNode extends NodeBase {
     private get azureAccount(): AzureAccountWrapper {
         return this.getTreeDataProvider<AppServiceDataProvider>().azureAccount;
     }
-}
\ No newline at end of file
+
+    public async createNewDeploymentSlot(): Promise<string> {
+        let slotName;
+        const slotNodes = await this.getChildren();
+        const slotLabels = slotNodes.map(node => {
+            return node.label;
+        });
+
+        slotName = await this.promptForSlotName(slotLabels);
+        if (!slotName) {
+            throw new UserCancelledError();
+        }
+        slotName = slotName.trim();
+        const newDeploymentSlot = {
+            name: slotName,
+            kind: this.site.kind,
+            location: this.site.location,
+            serverFarmId: this.site.serverFarmId
+        };
+        // if user has more slots than the service plan allows, Azure will respond with an error
+        await this.webSiteClient.webApps.createOrUpdateSlot(this.site.resourceGroup, util.extractSiteName(this.site), newDeploymentSlot, slotName);
+        return slotName;
+    }
+
+    protected get webSiteClient(): WebSiteManagementClient {
+        return new WebSiteManagementClient(this.azureAccount.getCredentialByTenantId(this.subscription.tenantId), this.subscription.subscriptionId);
+    }
+
+    protected async promptForSlotName(slotLabels: string[]): Promise<string | undefined> {
+        return await window.showInputBox({
+            prompt: 'Enter a unique name for the new deployment slot',
+            validateInput: (value: string) => {
+                value = value ? value.trim() : '';
+                if (!value.match(/^[a-z0-9\-]{1,60}$/ig)) {
+                    return 'Name should be 1-60 characters long and can only include alphanumeric characters and hyphens.';
+                }
+
+                // Can not have identical slot names OR production
+                if (value === 'production' || slotLabels.indexOf(value) !== -1) {

```

Comment:
Test case:
1. Create an Web App called "a-b".
2. Create another Web App called "a".
3. Now try to add a new deployment slot called "b" to Web App "a".

Now the new slot's domain name will be "a-b", which is already taken by the Web App in Step 1.

Reply:
Since we are going to revisit this (and the Web App Creator Wizard) when async InputBox validation is supported, let's just create an issue for now and use the current code.

Reply:
#166